include/RAJA/policy/openmp/forall.hpp

```diff
@@ -66,12 +67,15 @@ RAJA_INLINE void forall_impl(const omp_parallel_exec<InnerPolicy>&,
                              Iterable&& iter,
                              Func&& loop_body)
 {
-#pragma omp parallel
-  {
-    using RAJA::internal::thread_privatize;
-    auto body = thread_privatize(loop_body);
-    forall_impl(InnerPolicy{}, std::forward<Iterable>(iter), body.get_priv());
-  }
+    
+  RAJA::Region<RAJA::omp_parallel_region>([=, &iter](){
```

Comment:
@trws , I had to capture 'iter' by reference in order to get the code to compile. I was wondering if you might have a suggestion for an alternative implementation.

Reply:
I'm not sure what accounts for that.  One note is that the thread_privatize should be moved into the parallel region code, it's a property of launching the threads rather than specifically a loop.  Maybe send me something with the error, I'm not sure why there would be an issue with copying the iterable at this point in the code...

Reply:
The error was coming from trying to `forward` the iter, which was no longer a universal reference after being captured by value.

Reply:
Oh.  Then ~~std::forward(~~ iter ~~)~~ and capture it by value.

Reply:
Do you think that will have performance implications?

Reply:
Only where regions are invoked many times inside loops, and nested[2] was architected to help make that not matter. Perhaps more importantly, if it's re-worked as I mentioned above, the reductions should work for parallel code inside regions like this even outside of explicit RAJA worksharing loops.